### ISSUE TITLE
Typed/Racket broken in Git master.

### ISSUE THREAD

**Issue by racket-bug-submit** [samth/test-bugs#156]
###### Originally submitted by Ray Racine <ray.racine at gmail dot com> on: Tue Mar 13 10:32:01 -0400 2012

``` racket
Typed/Racket fails in DrRacket.
```
##### Steps to Reproduce:

``` racket
Build Racket from the git master branch.
Run the following program in DrRacket.

#lang typed/racket/base

(define x 3)
```

. . ../../usr/local/racket/collects/errortrace/errortrace-lib.rkt:44:0: syntax-rearm: expects type <syntax> as 1st argument, given: '(#<syntax (#%app register-type (quote-s...>); other arguments were: #<syntax (begin-for-syntax (#%app regi...>
##### Release:

```
5.2.900.1--2012-03-13(596a73d/g)
```
##### Environment:

```
unix "Linux ray.cdap2 3.2.7-1.fc16.x86_64 #1 SMP Tue Feb 21 01:40:47 UTC 2012 x86_64
 x86_64 x86_64 GNU/Linux" (x86_64-linux/3m) (get-display-depth) = 32
Human Language: english
(current-memory-use) 235658272
Links: (links) = (); (links #:user? #f) = (); (links #:root? #t) =
 (#<path:/code/racketlib>); (links #:user? #f #:root? #t) = ()


Collections:
("/home/ray/.racket/5.2.900.1/collects"
 (non-existent-path))
("/usr/local/racket/collects"
 ("rackunit" "xrepl" "handin-client" "mred" "readline" "rnrs" "tests" "file" "net" "s-exp"
 "trace" "r6rs" "mzscheme" ".gitignore" "meta" "htdp" "eopl" "db" "dynext" "typed"
 "slatex" "swindle" "schemeunit" "mysterx" "framework" "embedded-gui" "preprocessor"
 "redex" "test-box-recovery" "help" "gui-debugger" "version" "string-constants" "reader"
 "openssl" "plot" "wxme" "repo-time-stamp" "tex2page" "r5rs" "drscheme" "test-engine"
 "browser" "mzlib" "syntax" "hierlist" "racklog" "algol60" "slideshow" "honu" "drracket"
 "setup" "typed-racket" "planet" "picturing-programs" "sirmail" "parser-tools" "srfi"
 "html" "info-domain" "teachpack" "combinator-parser" "graphics" "icons" "scheme"
 "web-server" "scriblib" "frtime" "syntax-color" "racket" "unstable" "2htdp" "errortrace"
 "sgl" "defaults" "ffi" "typed-scheme" "deinprogramm" "raco" "lang" "config" "images"
 "data" "scribble" "mzcom" "texpict" "macro-debugger" "xml" "games" "plai" "make" "mrlib"
 "lazy" "handin-server" "profile" "scribbli!
 ngs" "at-exp" "compiler" "stepper" "launcher" "datalog"))

Computer Language: (("Determine language from source") (#(#t print mixed-fraction-e #f #t
 debug) (default) #() "#lang racket\n" #t #t))
```
###### This bug was converted from Gnats bug [12629](http://bugs.racket-lang.org/old/12629).

[anon-submit; ray.racine at gmail dot com]

**Comment by samth** [comment]
###### On Tue, 13 Mar 2012 10:47:45 -0400, samth at ccs dot neu dot edu wrote:

``` racket
This looks to me like a bug in errortrace.  Ray, can you change the
settings in the language dialog to "no debugging or profiling" and see
if the problem goes away?
```

**Comment by samth** [comment]
###### On Tue, 13 Mar 2012 11:15:24 -0400, samth at ccs dot neu dot edu wrote:

``` racket
[please maintain the CC list for the bug system]

On Tue, Mar 13, 2012 at 11:02 AM, Ray Racine <ray.racine@gmail.com> wrote:
> Overall I think my "guess" of the problem has to do with "entering
> namespaces".  Though interestingly "racket -I typed/racket -il xrepl" works
> fine, both DrRacket and Geiser both fail.

Right, DrRacket and Geiser are both much more sophisticated tools.

>
> With debug / profile off I see.

The below doesn't look like it's from DrRacket.

>
> [ray@ray ~]$ racket/gui/base is not available
>
>  === context ===
> /usr/local/racket/collects/racket/sandbox.rkt:94:0: sandbox-make-namespace
> /usr/local/racket/collects/racket/sandbox.rkt:430:0:
> make-evaluation-namespace
> /usr/local/racket/collects/racket/sandbox.rkt:666:0: make-evaluator*
> /usr/local/racket/collects/drracket/private/module-language.rkt:103:6:
> init-sandbox
> /usr/local/racket/collects/drracket/private/module-language.rkt:90:6:
> get-language-info method in .../module-language.rkt:86:4
> /usr/local/racket/collects/drracket/private/rep.rkt:924:6: submit-to-port?
> method in ...cket/private/rep.rkt:389:4
> /usr/local/racket/collects/framework/private/text.rkt:2153:4: on-local-char
> method in ...ork/private/text.rkt:1989:2
> /usr/local/racket/collects/mred/private/wxme/editor-canvas.rkt:444:2:
> on-char method in editor-canvas%
> /usr/local/racket/collects/racket/private/more-scheme.rkt:151:2:
> call-with-break-parameterization
> /usr/local/racket/collects/racket/private/more-scheme.rkt:268:2:
> call-with-exception-handler
> /usr/local/racket/collects/mred/private/wx/gtk/window.rkt:631:4:
> dispatch-on-char method in window%
> /usr/local/racket/collects/mred/private/wx/common/queue.rkt:445:6
> /usr/local/racket/collects/mred/private/wx/common/queue.rkt:485:32
> /usr/local/racket/collects/mred/private/wx/common/queue.rkt:631:3
>
> channel-put: expected argument of type <channel>; given: #f
>
>  === context ===
> /usr/local/racket/collects/racket/private/misc.rkt:168:2: channel-put
>
> racket/gui/base is not available
>
> On Tue, Mar 13, 2012 at 10:47 AM, Sam Tobin-Hochstadt <samth@ccs.neu.edu>
> wrote:
>>
>> This looks to me like a bug in errortrace.  Ray, can you change the
>> settings in the language dialog to "no debugging or profiling" and see
>> if the problem goes away?
>>
>> On Tue, Mar 13, 2012 at 10:32 AM,  <ray.racine@gmail.com> wrote:
>> >
>> > *** Description:
>> > Typed/Racket fails in DrRacket.
>> >
>> > *** How to repeat:
>> > Build Racket from the git master branch.
>> > Run the following program in DrRacket.
>> >
>> > #lang typed/racket/base
>> >
>> > (define x 3)
>> >
>> > . . ../../usr/local/racket/collects/errortrace/errortrace-lib.rkt:44:0:
>> > syntax-rearm: expects type <syntax> as 1st argument, given: '(#<syntax
>> > (#%app register-type (quote-s...>); other arguments were: #<syntax
>> > (begin-for-syntax (#%app regi...>
>>
>>
>>
>> --
>> sam th
>> samth@ccs.neu.edu
>
>
```

**Comment by samth** [comment]
###### On Tue, 13 Mar 2012 11:16:35 -0400, samth at ccs dot neu dot edu wrote:

``` racket
Major portions of the module system were just revised, which is almost
certainly causing this problem, esp. since it's not specific to TR.
For the moment, you may need to back up to an earlier version.

On Tue, Mar 13, 2012 at 11:10 AM, Ray Racine <ray.racine@gmail.com> wrote:
> A couple of other things on the issue.
>
> It seems the problem is not typed/racket specific after all.  I can see
> problems with #lang racket as well.
> Also I'm doing a really clean build from git.  i.e.,
>
> 1. git clean -f -x -d
> 2. ./configure
> 3. make
> 4. make install
>
> Which has worked reliably for me in the past.   I grabbed the lasted master
> to get the latest "remote places" code.  I can always go back to an earlier
> build point in git for now.  What I'm a bit puzzled by is that is should be
> breaking pretty much for everyone on the latest master, and if it is not,
> I'm doing something silly here on my end.
>
> I don't want to waster anyone's time chasing down dead end rabbit holes.
>
> Ray
>
> On Tue, Mar 13, 2012 at 10:47 AM, Sam Tobin-Hochstadt <samth@ccs.neu.edu>
> wrote:
>>
>> This looks to me like a bug in errortrace.  Ray, can you change the
>> settings in the language dialog to "no debugging or profiling" and see
>> if the problem goes away?
>>
>> On Tue, Mar 13, 2012 at 10:32 AM,  <ray.racine@gmail.com> wrote:
>> >
>> > *** Description:
>> > Typed/Racket fails in DrRacket.
>> >
>> > *** How to repeat:
>> > Build Racket from the git master branch.
>> > Run the following program in DrRacket.
>> >
>> > #lang typed/racket/base
>> >
>> > (define x 3)
>> >
>> > . . ../../usr/local/racket/collects/errortrace/errortrace-lib.rkt:44:0:
>> > syntax-rearm: expects type <syntax> as 1st argument, given: '(#<syntax
>> > (#%app register-type (quote-s...>); other arguments were: #<syntax
>> > (begin-for-syntax (#%app regi...>
>>
>>
>>
>> --
>> sam th
>> samth@ccs.neu.edu
>
>
```

**Comment by samth** [comment]
###### On Tue, 13 Mar 2012 11:23:07 -0400, samth at ccs dot neu dot edu wrote:

``` racket
[really, please maintain the CC list]

On Tue, Mar 13, 2012 at 11:20 AM, Ray Racine <ray.racine@gmail.com> wrote:
> No the below is being printed out by DrRacket (probably to stderr?) after I
> turn off errortrace and profile.

Does the program work in DrRacket?

>
>
> On Tue, Mar 13, 2012 at 11:15 AM, Sam Tobin-Hochstadt <samth@ccs.neu.edu>
> wrote:
>>
>> [please maintain the CC list for the bug system]
>>
>> On Tue, Mar 13, 2012 at 11:02 AM, Ray Racine <ray.racine@gmail.com> wrote:
>> > Overall I think my "guess" of the problem has to do with "entering
>> > namespaces".  Though interestingly "racket -I typed/racket -il xrepl"
>> > works
>> > fine, both DrRacket and Geiser both fail.
>>
>> Right, DrRacket and Geiser are both much more sophisticated tools.
>>
>> >
>> > With debug / profile off I see.
>>
>> The below doesn't look like it's from DrRacket.
>>
>> >
>> > [ray@ray ~]$ racket/gui/base is not available
>> >
>> >  === context ===
>> > /usr/local/racket/collects/racket/sandbox.rkt:94:0:
>> > sandbox-make-namespace
>> > /usr/local/racket/collects/racket/sandbox.rkt:430:0:
>> > make-evaluation-namespace
>> > /usr/local/racket/collects/racket/sandbox.rkt:666:0: make-evaluator*
>> > /usr/local/racket/collects/drracket/private/module-language.rkt:103:6:
>> > init-sandbox
>> > /usr/local/racket/collects/drracket/private/module-language.rkt:90:6:
>> > get-language-info method in .../module-language.rkt:86:4
>> > /usr/local/racket/collects/drracket/private/rep.rkt:924:6:
>> > submit-to-port?
>> > method in ...cket/private/rep.rkt:389:4
>> > /usr/local/racket/collects/framework/private/text.rkt:2153:4:
>> > on-local-char
>> > method in ...ork/private/text.rkt:1989:2
>> > /usr/local/racket/collects/mred/private/wxme/editor-canvas.rkt:444:2:
>> > on-char method in editor-canvas%
>> > /usr/local/racket/collects/racket/private/more-scheme.rkt:151:2:
>> > call-with-break-parameterization
>> > /usr/local/racket/collects/racket/private/more-scheme.rkt:268:2:
>> > call-with-exception-handler
>> > /usr/local/racket/collects/mred/private/wx/gtk/window.rkt:631:4:
>> > dispatch-on-char method in window%
>> > /usr/local/racket/collects/mred/private/wx/common/queue.rkt:445:6
>> > /usr/local/racket/collects/mred/private/wx/common/queue.rkt:485:32
>> > /usr/local/racket/collects/mred/private/wx/common/queue.rkt:631:3
>> >
>> > channel-put: expected argument of type <channel>; given: #f
>> >
>> >  === context ===
>> > /usr/local/racket/collects/racket/private/misc.rkt:168:2: channel-put
>> >
>> > racket/gui/base is not available
>> >
>> > On Tue, Mar 13, 2012 at 10:47 AM, Sam Tobin-Hochstadt
>> > <samth@ccs.neu.edu>
>> > wrote:
>> >>
>> >> This looks to me like a bug in errortrace.  Ray, can you change the
>> >> settings in the language dialog to "no debugging or profiling" and see
>> >> if the problem goes away?
>> >>
>> >> On Tue, Mar 13, 2012 at 10:32 AM,  <ray.racine@gmail.com> wrote:
>> >> >
>> >> > *** Description:
>> >> > Typed/Racket fails in DrRacket.
>> >> >
>> >> > *** How to repeat:
>> >> > Build Racket from the git master branch.
>> >> > Run the following program in DrRacket.
>> >> >
>> >> > #lang typed/racket/base
>> >> >
>> >> > (define x 3)
>> >> >
>> >> > . .
>> >> > ../../usr/local/racket/collects/errortrace/errortrace-lib.rkt:44:0:
>> >> > syntax-rearm: expects type <syntax> as 1st argument, given:
>> >> > '(#<syntax
>> >> > (#%app register-type (quote-s...>); other arguments were: #<syntax
>> >> > (begin-for-syntax (#%app regi...>
>> >>
>> >>
>> >>
>> >> --
>> >> sam th
>> >> samth@ccs.neu.edu
>> >
>> >
>>
>>
>>
>> --
>> sam th
>> samth@ccs.neu.edu
>
>
```

**Comment by samth** [comment]
###### On Tue, 13 Mar 2012 11:41:57 -0400, samth at ccs dot neu dot edu wrote:

``` racket
PLEASE maintain the CC list!  That probably means using "Reply All" in Gmail.

Ok, that's surely a DrRacket and/or Racket bug.

On Tue, Mar 13, 2012 at 11:27 AM, Ray Racine <ray.racine@gmail.com> wrote:
> No.
>
> If I turn off profile and trace.  I get the following behavior in DrRacket.
>
> #lang typed/racket
>
> (define x 3)
>
> 1. Syntax Check (works)
> 2. Run (works)
> 3. Type x <rtn> in the REPL prompt and I get the below being printed to
> stderr/stdout.
>
> Interestingly if I hit Run again I now start seeing "define-values x is
> already defined" type of error.  i.e, the namespace is not being destroyed
> and re-entered afresh.
>
> Change 'x' to 'z'.  Get the below, then further "Run"s error in DrR with 'z'
> already defined.
>
> Ray
>
>
> On Tue, Mar 13, 2012 at 11:23 AM, Sam Tobin-Hochstadt <samth@ccs.neu.edu>
> wrote:
>>
>> [really, please maintain the CC list]
>>
>> On Tue, Mar 13, 2012 at 11:20 AM, Ray Racine <ray.racine@gmail.com> wrote:
>> > No the below is being printed out by DrRacket (probably to stderr?)
>> > after I
>> > turn off errortrace and profile.
>>
>> Does the program work in DrRacket?
>>
>> >
>> >
>> > On Tue, Mar 13, 2012 at 11:15 AM, Sam Tobin-Hochstadt
>> > <samth@ccs.neu.edu>
>> > wrote:
>> >>
>> >> [please maintain the CC list for the bug system]
>> >>
>> >> On Tue, Mar 13, 2012 at 11:02 AM, Ray Racine <ray.racine@gmail.com>
>> >> wrote:
>> >> > Overall I think my "guess" of the problem has to do with "entering
>> >> > namespaces".  Though interestingly "racket -I typed/racket -il xrepl"
>> >> > works
>> >> > fine, both DrRacket and Geiser both fail.
>> >>
>> >> Right, DrRacket and Geiser are both much more sophisticated tools.
>> >>
>> >> >
>> >> > With debug / profile off I see.
>> >>
>> >> The below doesn't look like it's from DrRacket.
>> >>
>> >> >
>> >> > [ray@ray ~]$ racket/gui/base is not available
>> >> >
>> >> >  === context ===
>> >> > /usr/local/racket/collects/racket/sandbox.rkt:94:0:
>> >> > sandbox-make-namespace
>> >> > /usr/local/racket/collects/racket/sandbox.rkt:430:0:
>> >> > make-evaluation-namespace
>> >> > /usr/local/racket/collects/racket/sandbox.rkt:666:0: make-evaluator*
>> >> >
>> >> > /usr/local/racket/collects/drracket/private/module-language.rkt:103:6:
>> >> > init-sandbox
>> >> > /usr/local/racket/collects/drracket/private/module-language.rkt:90:6:
>> >> > get-language-info method in .../module-language.rkt:86:4
>> >> > /usr/local/racket/collects/drracket/private/rep.rkt:924:6:
>> >> > submit-to-port?
>> >> > method in ...cket/private/rep.rkt:389:4
>> >> > /usr/local/racket/collects/framework/private/text.rkt:2153:4:
>> >> > on-local-char
>> >> > method in ...ork/private/text.rkt:1989:2
>> >> > /usr/local/racket/collects/mred/private/wxme/editor-canvas.rkt:444:2:
>> >> > on-char method in editor-canvas%
>> >> > /usr/local/racket/collects/racket/private/more-scheme.rkt:151:2:
>> >> > call-with-break-parameterization
>> >> > /usr/local/racket/collects/racket/private/more-scheme.rkt:268:2:
>> >> > call-with-exception-handler
>> >> > /usr/local/racket/collects/mred/private/wx/gtk/window.rkt:631:4:
>> >> > dispatch-on-char method in window%
>> >> > /usr/local/racket/collects/mred/private/wx/common/queue.rkt:445:6
>> >> > /usr/local/racket/collects/mred/private/wx/common/queue.rkt:485:32
>> >> > /usr/local/racket/collects/mred/private/wx/common/queue.rkt:631:3
>> >> >
>> >> > channel-put: expected argument of type <channel>; given: #f
>> >> >
>> >> >  === context ===
>> >> > /usr/local/racket/collects/racket/private/misc.rkt:168:2: channel-put
>> >> >
>> >> > racket/gui/base is not available
>> >> >
>> >> > On Tue, Mar 13, 2012 at 10:47 AM, Sam Tobin-Hochstadt
>> >> > <samth@ccs.neu.edu>
>> >> > wrote:
>> >> >>
>> >> >> This looks to me like a bug in errortrace.  Ray, can you change the
>> >> >> settings in the language dialog to "no debugging or profiling" and
>> >> >> see
>> >> >> if the problem goes away?
>> >> >>
>> >> >> On Tue, Mar 13, 2012 at 10:32 AM,  <ray.racine@gmail.com> wrote:
>> >> >> >
>> >> >> > *** Description:
>> >> >> > Typed/Racket fails in DrRacket.
>> >> >> >
>> >> >> > *** How to repeat:
>> >> >> > Build Racket from the git master branch.
>> >> >> > Run the following program in DrRacket.
>> >> >> >
>> >> >> > #lang typed/racket/base
>> >> >> >
>> >> >> > (define x 3)
>> >> >> >
>> >> >> > . .
>> >> >> >
>> >> >> > ../../usr/local/racket/collects/errortrace/errortrace-lib.rkt:44:0:
>> >> >> > syntax-rearm: expects type <syntax> as 1st argument, given:
>> >> >> > '(#<syntax
>> >> >> > (#%app register-type (quote-s...>); other arguments were: #<syntax
>> >> >> > (begin-for-syntax (#%app regi...>
>> >> >>
>> >> >>
>> >> >>
>> >> >> --
>> >> >> sam th
>> >> >> samth@ccs.neu.edu
>> >> >
>> >> >
>> >>
>> >>
>> >>
>> >> --
>> >> sam th
>> >> samth@ccs.neu.edu
>> >
>> >
>>
>>
>>
>> --
>> sam th
>> samth@ccs.neu.edu
>
>
```

**Comment by racket-bug-submit** [comment]
###### On Tue, 13 Mar 2012 16:42:11 -0400, mflatt at racket-lang dot org assigned this bug to mflatt

A commit by mflatt@racket-lang.org has resolved this report
  http://git.racket-lang.org/plt/commit/99639c1999

**Comment by racket-bug-submit** [comment]
###### On Tue, 13 Mar 2012 16:42:11 -0400, mflatt at racket-lang dot org wrote:

A commit by mflatt@racket-lang.org has resolved this report: 99639c1999
